### ISSUE TITLE
Warning: Enable project-wide JavaScript/Typescript language features - large codebase

### ISSUE THREAD

**Issue by matyasmihalka** [vuejs/language-tools#3970]
After upgrade to v2 I am not able to get rid of this error saying `To enable project-wide JavaScript/TypeScript language features, exclude large folders with source files that you do not work on.` 

Note: it's on a large codebase with Vue 2.7. 

I noticed one interesting behaviour:
- If I restart VS Code and open a `.vue` file,  the language feature works fine for `.vue` files but not for `.ts`. 
- If I restart VS Code and open a `.ts` file, the language feature works fine for `.ts` files but not for `.vue`

I tried to add some more exclude patterns like the message suggests, but that didn't help. Also I would not like to go down this road as until now it worked fine. 

<img width="441" alt="image" src="https://github.com/vuejs/language-tools/assets/43904978/c6857639-7122-4b62-836f-507d00c8745b">

**Comment by davidmatter** [comment]
Is this one still an issue?

**Comment by matyasmihalka** [comment]
@davidmatter  Yes, it's still an issue. I just tried the v2 and I still receive the same warning as before and the above described behaviour is still the same.  

Otherwise I am working on `v1.8.27`, were it works as expected.

**Comment by KermanX** [comment]
Thank you for reporting. Please provide a minimal reproduction.

**Comment by matyasmihalka** [comment]
I am not sore how to do that in this case. This is happening on project where there are 100 000+ lines of code in Vue and seemingly it's connected with the size of the project. 
My experience is that on v2, if I open a Vue file the language feature works fine for that file even if I there is a warning, but if I switch to a .ts file, then it stops working. The behaviour is the same if I do the workflow in the opposite way.

**Comment by yllaw** [comment]
I am also experiencing this on a big Nuxt project.

Something I found is when I change `"allowJs": true` to   `"allowJs": false` the tsconfig in the `.nuxt`  generated build/dev folder it seems to fix itself.

**Comment by johnsoncodehk** [comment]
If you encounter the issue while using v3, please open a new issue and provide a reproduction repository.

On the other hand, you can use https://marketplace.visualstudio.com/items?itemName=johnsoncodehk.vscode-tsconfig-helper to check if your exclude patterns are set as expected.
